azurerm/resource_arm_autoscale_setting.go

```diff
@@ -0,0 +1,958 @@
+													}, true),
+													DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+												},
+												"value": {
+													Type:     schema.TypeInt,
+													Required: true,
+												},
+												"cooldown": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validateIso8601Duration(),
+												},
+											},
+										},
+										Set: resourceAzureRmAutoscaleDefaultHash,
+									},
+								},
+							},
+						},
+						"fixed_date": {
+							Type:          schema.TypeSet,
+							Optional:      true,
+							MaxItems:      1,
+							ConflictsWith: []string{"profile.recurrence"},
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"time_zone": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringInSlice(listTimeZoneNames(), false),
+									},
+									"start": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validateRFC3339Date,
+									},
+									"end": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validateRFC3339Date,
+									},
+								},
+							},
+							Set: resourceAzureRmAutoscaleDefaultHash,
+						},
+						"recurrence": {
+							Type:          schema.TypeSet,
+							Optional:      true,
+							MaxItems:      1,
+							ConflictsWith: []string{"profile.fixed_date"},
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"frequency": {
+										Type:     schema.TypeString,
+										Required: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											string(insights.Day),
+											string(insights.Hour),
+											string(insights.Minute),
+											string(insights.Month),
+											string(insights.None),
+											string(insights.Second),
+											string(insights.Week),
+											string(insights.Year),
+										}, true),
+										DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+									},
+									"schedule": {
+										Type:     schema.TypeSet,
+										Required: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"time_zone": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validation.StringInSlice(listTimeZoneNames(), false),
+												},
+												"days": {
+													Type:     schema.TypeList,
+													Required: true,
+													Elem: &schema.Schema{
+														Type: schema.TypeString,
+														ValidateFunc: validation.StringInSlice([]string{
+															"Monday",
+															"Tuesday",
+															"Wednesday",
+															"Thursday",
+															"Friday",
+															"Saturday",
+															"Sunday",
+														}, true),
```

Comment:
Is it case sensitive? If not, we may add `DiffSuppressFunc: ignoreCaseDiffSuppressFunc` here.